File: TheHuxley/817 - Chuva (OBI).py
# Erro no código
def max(a, b):
    if(a > b):
        return a
    else:
        return b

size = int(input())
hight = []
minleft = []
minright = []
mindef = []

sub = 0

for i in range(size):
    v = int(input())
    hight.append(v)
minleft[0] = hight[0]

for i in range(1, size, 1):
    minleft[i] = max(minleft[i - 1], hight[i])
minright[size - 1] = hight[size - 1]

for i in range(size - 2, 0, 1):
    minright[i] = max(minright[i + 1], hight[i])

for i in range(size):
    mindef[i] = -max(-minright[i], -minleft[i])

for i in range(size):
    if mindef[i] > hight[i]:
        sub += 1

print(sub)

File: TheHuxley/817 - Chuva (OBI).c
#include <stdio.h>
#include <stdlib.h>

int max(int a, int b)
{
    if (a > b)
    {
        return a;
    }
    return b;
}

int main()
{
    int size;
    int i, b, c;

    scanf("%d", &size);

    int hight[size], sub = 0;
    int minleft[size], minright[size], mindef[size];

    for (i = 0; i < size; i++)
    {
        scanf("%d", &hight[i]);
    }
    minleft[0] = hight[0];
    for (i = 1; i < size; i++)
    {
        minleft[i] = max(minleft[i - 1], hight[i]);
    }
    minright[size - 1] = hight[size - 1];
    for (i = size - 2; i >= 0; i--)
    {
        minright[i] = max(minright[i + 1], hight[i]);
    }
    for (i = 0; i < size; i++)
    {
        mindef[i] = -max(-minright[i], -minleft[i]);
    }
    for (i = 0; i < size; i++)
    {
        if (mindef[i] > hight[i]) {
            sub++;
        }
    }

    printf("%d\n", sub);

    return 0;
}
